data/keystore/src/main/java/tech/pegasys/artemis/bls/keystore/KeyStoreLoader.java

```
@@ -0,0 +1,104 @@
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.bls.keystore;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.exc.InvalidTypeIdException;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import tech.pegasys.artemis.bls.keystore.model.KdfParam;
+import tech.pegasys.artemis.bls.keystore.model.KeyStoreData;
+
+/** Provide utility methods to load/store BLS KeyStore from json format */
+public class KeyStoreLoader {
+  private static final Logger LOG = LogManager.getLogger();
+  private static final ObjectMapper OBJECT_MAPPER =
+      new ObjectMapper().registerModule(new KeyStoreBytesModule());
+
+  public static KeyStoreData loadFromFile(final Path keystoreFile)
+      throws KeyStoreValidationException {
+    checkNotNull(keystoreFile, "KeyStore path cannot be null");
+
+    try {
+      final KeyStoreData keyStoreData =
+          OBJECT_MAPPER.readValue(keystoreFile.toFile(), KeyStoreData.class);
+      keyStoreData.validate();
+      return keyStoreData;
+    } catch (final KeyStoreValidationException e) {
+      throw new KeyStoreValidationException("Error in parsing keystore: " + e.getMessage(), e);
```

Why are we re-throwing the same exception type?  Seems like we should just not catch it and let the original bubble up.

---

The initial idea was to prefix "Error in parsing" but you are right, we should bubble up original exception.